jabgui/src/main/java/org/jabref/gui/walkthrough/declarative/NavigationPredicate.java

```diff
@@ -0,0 +1,212 @@
+                    button.removeEventFilter(ActionEvent.ACTION, decoratedAction);
+                };
+            }
+
+            return () -> targetNode.setOnMouseClicked(onClicked);
+        };
+    }
+
+    static NavigationPredicate onHover() {
+        return (targetNode, beforeNavigate, onNavigate) -> {
+            EventHandler<? super MouseEvent> onEnter = targetNode.getOnMouseEntered();
+            targetNode.setOnMouseEntered(decorate(beforeNavigate, onEnter, onNavigate));
+
+            Optional<MenuItem> item = resolveMenuItem(targetNode);
+            if (item.isPresent()) {
+                throw new IllegalArgumentException("onHover cannot be used with MenuItems");
+            }
+
+            return () -> targetNode.setOnMouseEntered(onEnter);
+        };
+    }
+
+    static NavigationPredicate onTextInput() {
+        return (targetNode, beforeNavigate, onNavigate) -> {
+            if (targetNode instanceof TextInputControl textInput) {
+                ChangeListener<String> listener = (_, _, newText) -> {
+                    if (!newText.trim().isEmpty()) {
+                        beforeNavigate.run();
+                        onNavigate.run();
+                    }
+                };
+                textInput.textProperty().addListener(listener);
+                return () -> textInput.textProperty().removeListener(listener);
+            }
+            throw new IllegalArgumentException("onTextInput can only be used with TextInputControl");
+        };
+    }
+
+    static NavigationPredicate manual() {
+        return (_, _, _) -> () -> {
+        };
+    }
+
+    static NavigationPredicate auto() {
+        return (_, _, onNavigate) -> {
+            onNavigate.run();
+            return () -> {
+            };
+        };
+    }
+
+    private static Optional<MenuItem> resolveMenuItem(Node node) {
+        if (!(node instanceof ContextMenuContent)
+                && Stream.iterate(node.getParent(), Objects::nonNull, Parent::getParent)
+                         .noneMatch(ContextMenuContent.class::isInstance)) {
+            return Optional.empty();
+        }
+
+        return Window.getWindows().stream()
+                     .map(Window::getScene)
+                     .filter(Objects::nonNull)
+                     .map(scene -> scene.lookup(".mainMenu"))
+                     .filter(MainMenu.class::isInstance)
+                     .map(MainMenu.class::cast)
+                     .flatMap(menu -> menu.getMenus().stream())
+                     .flatMap(topLevelMenu -> topLevelMenu.getItems().stream())
+                     .filter(menuItem -> Optional.ofNullable(menuItem.getGraphic())
+                                                 .map(graphic -> graphic.equals(node)
+                                                         || Stream.iterate(graphic, Objects::nonNull, Node::getParent)
+                                                                  .anyMatch(cm -> cm.equals(node)))
```

Comment:
If I am not wrong on the expansion:
```suggestion
                     .filter(menuItem -> Optional.ofNullable(menuItem.getGraphic())
                                                 .map(graphic -> graphic.equals(node)
                                                         || Stream.iterate(graphic, Objects::nonNull, Node::getParent)
                                                                  .anyMatch(contextMenu -> contextMenu.equals(node)))
```